### ISSUE TITLE
GPU bandwidth is 35% (5.3 GB/s) of the theoretical 16 GB/s

### ISSUE THREAD

**Issue by arrmansa** [arrmansa/Gpt-Neo-Limited-Vram-Cuda#1]
and the CPU usage seems to max out a single core.
This indicates that it may be much faster to transfer data to gpu in multiple threads/processes, but I'm not sure how this could be done.
5.3 GB/s comes from 50 loads of the (4.9Gb) model in 46 seconds

**Comment by arrmansa** [comment]
The bandwidth on google colaboratory is higher at 6.4Gb/s
(50 loads in 38 seconds)

**Comment by arrmansa** [comment]
This has been fixed by using .detach() in the new repository
